src/lib.rs

```diff
@@ -44,6 +45,34 @@ pub fn guess(xdr_base64: String) -> Vec<String> {
         .collect()
 }
 
+/// Decodes the XDR into JSON.
+///
+/// Accepts a XDR base64 string.
+///
+/// Returns a JSON string.
+///
+/// Unstable: The API of this function is unstable and will likely be changed to
+/// return a JsValue instead of a JSON string.
+#[wasm_bindgen]
+pub fn decode_stream(type_variant: String, xdr_base64: String) -> Result<String, String> {
```

Comment:
Can we return the vec as a vec? We can use Vec's across the JS<>WASM boundary, and so I don't think there's a need to have the array encoded. On other functions we return Vec's.
```suggestion
pub fn decode_stream(type_variant: String, xdr_base64: String) -> Result<Vec<String>, String> {
```

Reply:
I agree that this would be better interface. However, we still need to decode each string in json so the caller would need to do more to convert the array into a JS array. I did some experiments with using JsValue and wasm-serde, which would result in handling this for the caller. So I suggest we don't stabilize this API and I return to this. The JsValue route ties the implementation more closely with Js, but it did result in a smaller binary. (I also need to update the xdr generation if we go that route.

Reply:
On second thought it isn't too much to ask the caller to handle it so perhaps for the clarity of this returning a vec we go for it.

Reply:
If we find a way to use JsValue with good tradeoffs, then won't it be used in all the functions, not just this one? So I don't think we need to mark this one as especially unstable vs the rest of the lib.